src/model/reference.ts

```diff
@@ -56,39 +58,70 @@ export class ScannedReference<K extends keyof FeatureKindMap> extends
   // specific kind (e.g. resolve a PolymerElement rather than just a Class).
   resolveWithKind<DK extends keyof FeatureKindMap>(
       document: Document, kind: DK): Reference<FeatureKindMap[DK]> {
-    const features = document.getFeatures(
-        {imported: true, externalPackages: true, kind, id: this.identifier});
+    let feature: undefined|FeatureKindMap[DK];
     const warnings = [...this.warnings];
-    if (this.sourceRange) {
-      if (features.size === 0) {
-        let message = `Could not resolve reference to ${this.kind}`;
-        if (kind === 'behavior') {
-          message += `. Is it annotated with @polymerBehavior?`;
+
+    const binding = this.astPath.scope.getBinding(this.identifier);
+    if (binding !== undefined) {
```

Comment:
We should file an issue to make this more principled. We can know at scanning time whether a binding was to a locally defined name, and only look up via scope analysis in that case. We can also attempt both look up styles and report when resolution is ambiguous, though you pointed out that this would be verbose for classes, which always add themselves to the global identifier map. We should also file an issue to fix that and only add to namespaces if annotated with `@memberof`, etc.

Reply:
Filed https://github.com/Polymer/polymer-analyzer/issues/918 and https://github.com/Polymer/polymer-analyzer/issues/917